crates/env_filter/src/parser.rs

```diff
@@ -66,22 +107,39 @@ pub(crate) fn parse_spec(spec: &str) -> (Vec<Directive>, Option<FilterOp>) {
         }
     }
 
-    let filter = filter.and_then(|filter| match FilterOp::new(filter) {
-        Ok(re) => Some(re),
-        Err(e) => {
-            eprintln!("warning: invalid regex filter - {}", e);
-            None
-        }
-    });
+    let filter = filter
+        .map(|filter| match FilterOp::new(filter) {
+            Ok(re) => Ok(Some(re)),
+            Err(e) => err_or_print(
+                skip_errors,
+                format!("warning: invalid regex filter - {}", e),
+            )
+            .map(|_| None),
+        })
+        .transpose()?
+        .flatten();
```

Comment:
This is getting to be too complex of combinators.  I'm assuming there is a more straightforward imperative way of writing this.  Maybe refactoring the original code could also be its own commit, paving the way for the added complexity of this

Reply:
With the new `ParseResult` this code seems much cleaner.